File: java/src/test/java/problems1_100/JumpGame2Test.java
package problems1_100;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class JumpGame2Test extends SampledTest {
    @Override
    protected String getSampleFilename() {
        return "jump-game-ii.json";
    }

    @Test
    public void test() {
        for (JumpGame2 solution : JumpGame2.values()) {
            for (JumpGame2Sample sample : getSamples(JumpGame2Sample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.jump(sample.input.nums))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/JumpGame2Test.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class JumpGame2Test : SampledTest() {
    override fun getSampleFilename(): String = "jump-game-ii.json"

    @Test
    fun test() =
        JumpGame2.entries.forEach { solution ->
            getSamples<Array<JumpGame2Sample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.jump(input.nums))
                    .isEqualTo(output)
            }
        }
}
